### PR TITLE
[Scheduler] User friendly cron expressions

### PR DESCRIPTION
**Pull request by SferaDev**
### :memo: Implementation

- In the sync rules wizard, when inserting a valid cron expression, we previsualize the result. If it's incorrect we have the default "Select frequency" placeholder.

- We should probably update the options we suggest in the dropdown to match the same auto generated string (but I guess we should think before which cron expressions are useful to the clients ie: Every day at 11PM from Monday to Friday)

### :art: Screenshots

image

image

image

image

image

### :fire: Is there anything the reviewer should know to test it?

- The library supports multiple locales, not sure if we want to enable that feature yet (they support more languages than us and could lead to mixed translations UI)

- Added as string interpolation, also tested adding tooltips but not sure what's the best approach.

image

### CODE REVIEW

File: src/components/rules-creation-page/steps/SchedulerStep.jsx [hunk 1]
```diff
@@ -56,7 +57,11 @@ const SchedulerStep = ({ syncRule, onChange }) => {
             value: selectedCron.value || "",
             component: DropDown,
             props: {
-                hintText: i18n.t("Select frequency"),
+                hintText: i18n.t(
+                    isValidCronExpression(syncRule.frequency)
+                        ? cronstrue.toString(syncRule.frequency)
+                        : "Select frequency"
```
Comment (tokland) [inline src/components/rules-creation-page/steps/SchedulerStep.jsx]: I'd keep the `i18n.t` here.

File: src/components/rules-creation-page/steps/SaveStep.jsx [hunk 2]
```diff
@@ -107,7 +109,12 @@ const SaveStep = ({ d2, syncRule, classes, onCancel, snackbar }) => {
 
                 <LiEntry label={i18n.t("Scheduling enabled")} value={syncRule.enabled.toString()} />
 
-                <LiEntry label={i18n.t("Frequency")} value={syncRule.frequency} />
+                {isValidCronExpression(syncRule.frequency) && (
+                    <LiEntry
+                        label={i18n.t("Frequency")}
+                        value={`${cronstrue.toString(syncRule.frequency)} (${syncRule.frequency})`}
```
Comment (tokland) [inline src/components/rules-creation-page/steps/SaveStep.jsx]: This snippet seens quite generic and it's repeated in the code. Maybe we can move it as a method of `SyncRule`?

File: src/components/rules-list-page/SyncRulesPage.jsx [hunk 3]
```diff
@@ -67,20 +69,34 @@ class SyncRulesPage extends React.Component {
                     .map(e => e.name)
                     .join(", "),
         },
-        { name: "frequency", text: i18n.t("Frequency"), sortable: true },
+        {
+            name: "frequency",
+            text: i18n.t("Frequency"),
+            sortable: true,
+            getValue: ({ frequency }) =>
+                isValidCronExpression(frequency)
```
Comment (tokland) [inline src/components/rules-list-page/SyncRulesPage.jsx]: Same than before, does it make sense to have `syncRule.isValidFrequency()` or something like that?

File: src/components/rules-creation-page/steps/SchedulerStep.jsx (hunk 1 above)
Comment (SferaDev) [reply src/components/rules-creation-page/steps/SchedulerStep.jsx]: See https://github.com/EyeSeeTea/metadata-synchronization/pull/134/commits/d4eebaf7a6ac2897e547c59cf1986b291d022208

File: src/components/rules-creation-page/steps/SaveStep.jsx (hunk 2 above)
Comment (SferaDev) [reply src/components/rules-creation-page/steps/SaveStep.jsx]: See https://github.com/EyeSeeTea/metadata-synchronization/pull/134/commits/d4eebaf7a6ac2897e547c59cf1986b291d022208

File: src/components/rules-list-page/SyncRulesPage.jsx (hunk 3 above)
Comment (SferaDev) [reply src/components/rules-list-page/SyncRulesPage.jsx]: See https://github.com/EyeSeeTea/metadata-synchronization/pull/134/commits/d4eebaf7a6ac2897e547c59cf1986b291d022208

**Pull request merged**